lib/facility_factory.rb

```diff
@@ -0,0 +1,54 @@
+
+
+class FacilityFactory
+
+  def initialize
+  end
+
+  def create_facility(input_api)
+    co_total_facility_array = []
+    ny_total_facility_array = []
+    mo_total_facility_array = []
+
+    if input_api.first[:state] == "CO"
+      input_api.each do |facility_raw_info|
+        input_hash = {}
+        input_hash[:name] = facility_raw_info[:dmv_office]
+        input_hash[:address] = "#{facility_raw_info[:address_li]}, #{facility_raw_info[:address__1]}, #{facility_raw_info[:city]}, #{facility_raw_info[:state]} #{facility_raw_info[:zip]}"
+        input_hash[:phone] = facility_raw_info[:phone]
+        input_hash[:hours] = facility_raw_info[:hours]
+        co_total_facility_array << Facility.new(input_hash)
+      end
+    return co_total_facility_array
+  
```

Comment:
I like that you're explicitly returning the array after it's been filled to avoid your method continuing to run.